server/routes/user.js

```diff
@@ -1,8 +1,76 @@
 const express = require("express");
 const router = express.Router();
-const User = require("../controllers/user");
+const bcrypt = require("bcryptjs");
+const jwt = require("jsonwebtoken");
+const config = require("../config/dev");
+const { check, validationResult } = require("express-validator");
 
-router.post("/register", User.register);
-router.post("/auth", User.authenticate);
+const User = require("../models/user");
 
-module.exports = router;
\ No newline at end of file
+// router.post("/register", User.register);
+// router.post("/auth", User.authenticate);
+
+// @route   POST api/user
+// @desc    Registers a new user if the email is unique, and returns the user's token
+// @access  Public
+router.post(
+  "/",
+  // Validate input data
+  [
+    check("email", "Please include a valid email").isEmail(),
+    check("password", "Password must be 6 or more characters").isLength({
+      min: 6
+    })
+  ],
+  async (req, res) => {
+    const errors = validationResult(req);
+    if (!errors.isEmpty()) {
+      return res.status(400).json({ errors: errors.array() });
+    }
+
+    const { email, password } = req.body;
+
+    try {
+      // See if user exists
+      let user = await User.findOne({ email });
```

Comment:
Do we still need this when we put unique: true in the model? Just wondering

Reply:
Yea, we want to handle that error here, so we can return our own custom error message.